politeiad/backendv2/tstorebe/plugins/comments/cmds.go

```diff
@@ -441,9 +441,21 @@ func (p *commentsPlugin) cmdNew(token []byte, payload string) (string, error) {
 		return "", err
 	}
 
+	// Ensure no extra data provided if not allowed
+	if !p.allowExtraData && (n.ExtraData != "" || n.ExtraDataHint != "") {
+		return "", backend.PluginError{
+			PluginID:     comments.PluginID,
+			ErrorCode:    uint32(comments.ErrorCodeExtraDataNotAllowed),
+			ErrorContext: "comment extra data is not allowed",
+		}
+	}
+
 	// Verify signature
 	msg := strconv.FormatUint(uint64(n.State), 10) + n.Token +
 		strconv.FormatUint(uint64(n.ParentID), 10) + n.Comment
+	if p.allowExtraData {
```

Comment:
You don't need this check.  Empty strings won't impact the signature and it's cleaner without an extra code path.

```suggestion
	msg := strconv.FormatUint(uint64(n.State), 10) + n.Token +
		strconv.FormatUint(uint64(n.ParentID), 10) + n.Comment +
		n.ExtraData + n.ExtraDataHint
```

Reply:
agreed